src/scenes/Projects/Overview/ProjectOverview.tsx

```diff
@@ -88,180 +96,195 @@ export const ProjectOverview: FC = () => {
-                  )
-                ) : (
-                  <Skeleton width="40%" />
+                    <Skeleton width="40%" />
+                  )}
+                </Typography>
+              </Grid>
+              <Grid item>
+                {data?.project.engagements.canCreate && (
+                  <Tooltip title={`Add ${engagementTypeLabel} Engagement`}>
+                    <Fab
+                      color="error"
+                      aria-label={`Add ${engagementTypeLabel} Engagement`}
+                      onClick={
+                        data.project.__typename === 'TranslationProject'
+                          ? createLanguageEngagement
+                          : createInternshipEngagement
+                      }
+                    >
+                      <Add />
+                    </Fab>
+                  </Tooltip>
                 )}
-              </Typography>
+              </Grid>
             </Grid>
-            <Grid item>
-              {data?.project.engagements.canCreate && (
-                <Tooltip title={`Add ${engagementTypeLabel} Engagement`}>
-                  <Fab
-                    color="error"
-                    aria-label={`Add ${engagementTypeLabel} Engagement`}
-                  >
-                    <Add />
-                  </Fab>
-                </Tooltip>
-              )}
-            </Grid>
-          </Grid>
-          {data?.project?.engagements?.items?.map((engagement) =>
-            engagement.__typename === 'LanguageEngagement' ? (
-              <LanguageEngagementListItemCard
-                key={engagement.id}
-                projectId={projectId}
-                {...engagement}
-              />
-            ) : engagement.__typename === 'InternshipEngagement' ? (
-              <InternshipEngagementListItemCard
-                key={engagement.id}
-                projectId={projectId}
-                {...engagement}
-              />
-            ) : null
-          )}
-        </div>
-      )}
-    </main>
+            {data?.project?.engagements?.items?.map((engagement) =>
+              engagement.__typename === 'LanguageEngagement' ? (
+                <LanguageEngagementListItemCard
+                  key={engagement.id}
+                  projectId={projectId}
+                  {...engagement}
+                />
+              ) : engagement.__typename === 'InternshipEngagement' ? (
+                <InternshipEngagementListItemCard
+                  key={engagement.id}
+                  projectId={projectId}
+                  {...engagement}
+                />
+              ) : null
+            )}
+          </div>
+        )}
+      </main>
+      <CreateLanguageEngagement
+        projectId={projectId}
+        {...createLanguageEngagementState}
+      />
+      <CreateInternshipEngagement
+        projectId={projectId}
+        {...createInternshipEngagementState}
+      />
```

Comment:
These don't necessarily have to be here at the component root. They already portal to another spot completely in the DOM. I think it would make more sense to have these next to the add button.
That also means we don't have to indent the entire JSX to add a Fragment.

Also we don't need to mount both - only one or the other.

Reply:
Looks like the fragment is already there for UploadProjectFiles, but I'll move the dialogs next to the button.

Reply:
Ah ok cool